game.js

```diff
@@ -0,0 +1,90 @@
+var Game = function(board){
+  this.board = board;
+};
+
+Game.prototype = {
+  // tests all activated tile(s) if it matches with the clicked tile
+  findAnyMatch: function(clicked_tile){
+    var matched_tile;
+    this.board.allActivatedTiles().forEach(
+        function(tile){
+          if (tile.id !== clicked_tile.id && tile.value_matches(clicked_tile)){
+            matched_tile = tile;
+          }
+        }
+    );
+    return matched_tile;
+  },
+
+  findAndRecordAnyMatch: function(clicked_tile){
+     var matched_tile = this.findAnyMatch(clicked_tile);
+     if (matched_tile !== undefined){
```

Comment:
Unless you care about the value being `undefined` specifically, I would simplify this to `if (!matched_tile)`.